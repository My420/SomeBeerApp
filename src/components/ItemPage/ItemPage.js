import React from 'react';
import PropTypes from 'prop-types';
import styles from './ItemPage.module.scss';
import Main from '../Main/Main';
import Container from '../Container/Container';
import CatalogBeerPage from '../BeerPage/BeerPage';

const ItemPage = ({ match }) => {
  // eslint-disable-next-line no-console
  console.log('render ==================== ItemPage');
  const { id } = match.params;

  return (
    <Main className={styles.main}>
      <Container className={styles.container}>
        <CatalogBeerPage id={id} />
      </Container>
    </Main>
  );
};

ItemPage.propTypes = {
  match: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default ItemPage;
