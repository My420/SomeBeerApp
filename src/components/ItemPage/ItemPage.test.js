import React from 'react';
import { shallow } from 'enzyme';
import ItemPage from './ItemPage';

describe('test <ItemPage /> component', () => {
  const match = {
    params: { id: 22 }
  };

  const wrapper = shallow(<ItemPage match={match} />);
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render CatalogBeerPage component with correct props', () => {
    const component = wrapper.find('Connect(BeerPage)');
    expect(component).toHaveLength(1);
    expect(component.prop('id')).toBe(match.params.id);
  });
});
