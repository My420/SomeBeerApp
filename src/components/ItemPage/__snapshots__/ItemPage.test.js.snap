// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <ItemPage /> component should match snapshot 1`] = `
<Main
  className="main"
>
  <Container
    className="container"
  >
    <Connect(BeerPage)
      id={22}
    />
  </Container>
</Main>
`;
